services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -34,7 +34,7 @@ class ProjectService(projectRepository: ProjectRepository, projectVersioning: Pr
         active = true
       )
     projectVersioning.createRepository(project).flatMap {
-      case Left(exception)       => Future.successful(Left(exception))
+      case Left(exception)       => Future.failed(exception)
```

Comment:
What is the point of returning `Either[A, B]` if we always return Right?
What problem you are trying to solve with this fix?

Reply:
Changed returning on ```Future[ProjectId]```

Reply:
Maybe you should have fix the controller? :)
It can return some custom response code in case of `VersioningException`

Reply:
Changed on matching in controller